Update _encoders.py

krishnakatyal opened the pull request:
**Summary**
Remove the word dense array as the output is sparse not dense.
A sparse matrix has most of the elements zero where as a  the dense one is opposite of sparse ,this creates a contradiction .
The returned matrix cant be both dense and sparse.

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

Fixes a mistake in documentation of dense and sparse array.

krishnakatyal committed:
Update _encoders.py

Remove the word dense array as the output is sparse not dense.
A sparse matrix has most of the elements zero where as a  the dense one is opposite of sparse ,this creates a contradiction .
The returned matrix cant be both dense and sparse.

rth commented:
There is a `sparse` (default to True) parameter in OneHotEncoder that specifies whether the output is sparse. It can be dense..

krishnakatyal commented:
@rth using contradictory words , the documentation becomes difficult to understand
i beginner reading will get confused

rth commented:
Feel free to improve the wording, but that estimator does return a sparse matrix or dense array, depending on the value of the `sparse` parameter.

krishnakatyal committed:
Update _encoders.py (#1)

The wording was confusing i improved the existing wording. The explanation of sparse and when the output will be dense or sparse is also now provided

krishnakatyal commented:
@rth I have made a new pull request regarding this issue.
